sdk/program/src/pubkey.rs

```diff
@@ -151,11 +170,12 @@ pub fn bytes_are_curve_point<T: AsRef<[u8]>>(_bytes: T) -> bool {
 }
 
 impl Pubkey {
+    #[deprecated(
+        since = "1.14.14",
+        note = "Please use 'Pubkey::from' or 'Pubkey::try_from' instead"
+    )]
     pub fn new(pubkey_vec: &[u8]) -> Self {
-        Self(
-            <[u8; 32]>::try_from(<&[u8]>::clone(&pubkey_vec))
-                .expect("Slice must be the same length as a Pubkey"),
-        )
+        Self(<[u8; 32]>::try_from(pubkey_vec).expect("Slice must be the same length as a Pubkey"))
```

Comment:
```suggestion
        Self::try_from(pubkey_vec).expect("Slice must be the same length as a Pubkey")
```

dereplicode (i think)